### ISSUE TITLE
Please reconsider the GPL decision

### ISSUE THREAD

**Issue by KrishnaPG** [deepstreamIO/deepstream.io#298]
I see that the project has moved from Apache to AGPL.

It is a too-restrictive licence and forces all the combined code to become AGPL.

Given that many components this project has to work with and rely upon (including the nodejs itself) are liberal and non-GPL style of licenses (variants of Apache, BSD, MIT etc.), it is only fair that this uses more permissive license than "all or nothing" style of AGPL.

Or, you can pick a mid-point between the two and use MPL (mozilla public license) - that will keep the modifications to this project open, while respecting the ownership of other components to use their own license. AGPL does not have that kind of respect for other licenses.

**Comment by yasserf** [comment]
Hey Krishna, thank you for your input.

We have put alot of effort into making deepstream as extensible as possible. What this means is if you decide to write your own custom logic for authentication, permissions, logging, cache, clustering or storage you can provide a custom plugin that does not need to be open-sourced, although in the spirit of open-source we would very happy if it was.

Since deepstream is meant to be run as a standalone server using it within your stack does not force you to open your codebase. You can see examples of this with other backend services such as rethinkDB and mongoDB.

We have also kept all our client libraries under Apache 2.0 to allow them to be customized in anyway required to suit your applications.

In summary, as long as you don't make any changes to the deepstream server core logic that you **don't** want to share you don't have to license software that uses deepstream with AGPL and can license that software under any license you want.

**Comment by KrishnaPG** [comment]
Understood. The parallel with MongoDB and RethinkDB, along with your clarification that the client libraries are kept under Apache 2, makes your view clear.
